### PR TITLE
Re-enable c_bindings example

### PR DESCRIPTION
**Pull request by havogt**
- improve GridToolConfig.cmake
- fix is_gpu_ptr()
- propagate CUDA_ARCH in all examples
- fix separation of CUDA/non-CUDA compilation in stencil_computation example

### CODE REVIEW

**Comment by havogt** [comment]
launch jenkins

File: include/gridtools/storage/info.hpp [hunk 1]
```diff
@@ -44,7 +44,9 @@ namespace gridtools {
 
             template <int... Dims, int... LayoutArgs, class Array>
             constexpr Array make_strides(layout_map<LayoutArgs...> layout, uint_t align, Array const &lengths) {
+#if not defined(__CUDA_ARCH__) || (defined(__CUDACC_VER_MAJOR__) && __CUDACC_VER_MAJOR__ >= 10)
```
[the remark this answers is not in the document]
Comment (havogt) [reply include/gridtools/storage/info.hpp:47]: The problem was when I tried to compile the c_bindings examples with CUDA 9.2. Don't ask me what's the problem. Very weird error. I'll remove it as we drop support for CUDA 9.x.

**Comment by havogt** [comment]
launch mpitest

**Comment by havogt** [comment]
launch jenkins

File: include/gridtools/storage/info.hpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (anstaf) [reply include/gridtools/storage/info.hpp]: Please, either add a comment or remove the assertion completely.

[the remark this answers is not in the document]
Comment (havogt) [reply include/gridtools/storage/info.hpp]: I didn't touch the assertion.

[the remark this answers is not in the document]
Comment (havogt) [reply include/gridtools/storage/info.hpp]: I removed the ifdef already

**Pull request merged**